src/clj/muuntaja/json.clj

```diff
@@ -67,61 +69,78 @@
   "Create a Jackson Databind module to support Clojure datastructures.
 
   See make-mapper docstring for the documentation of the options."
-  [{:keys [keywordize? encoders]}]
+  [{:keys [keywordize? encoders date-format]}]
   (doto (SimpleModule. "Clojure")
     (.addDeserializer java.util.List (PersistentVectorDeserializer.))
     (.addDeserializer java.util.Map (PersistentHashMapDeserializer.))
     (.addSerializer clojure.lang.Keyword (KeywordSerializer. false))
     (.addSerializer clojure.lang.Ratio (RatioSerializer.))
     (.addSerializer clojure.lang.Symbol (SymbolSerializer.))
-    (.addSerializer java.util.Date (DateSerializer.))
     (.addKeySerializer clojure.lang.Keyword (KeywordSerializer. true))
+    (.addSerializer java.util.Date (if date-format
+                                     (DateSerializer. date-format)
+                                     (DateSerializer.)))
     (as-> module
         (doseq [[cls encoder-fn] encoders]
           (.addSerializer module cls (FunctionalSerializer. encoder-fn))))
     (cond->
-      ;; This key deserializer decodes the map keys into Clojure keywords.
-      keywordize? (.addKeyDeserializer Object (KeywordKeyDeserializer.)))))
+        ;; This key deserializer decodes the map keys into Clojure keywords.
+        keywordize? (.addKeyDeserializer Object (KeywordKeyDeserializer.)))))
 
 (defn ^ObjectMapper make-mapper
   "Create an ObjectMapper with Clojure support.
 
   The optional first parameter is a map of options. The following options are
   available:
-
+  Encodin options: 
+  :pretty           -- set to true use Jacksons pretty-printing defaults
+  :escape-non-ascii -- set to true to escape non ascii characters
+  :date-format      -- string for custom date formatting. If not set will use default \"yyyy-MM-dd'T'HH:mm:ss'Z'\"
   :encoders     --  a map of custom encoders where keys should be types and values
                     should be encoder functions
-  :keywordize?  --  set to true to convert map keys into keywords (default: false)
-
   Encoder functions take two parameters: the value to be encoded and a
   JsonGenerator object. The function should call JsonGenerator methods to emit
-  the desired JSON."
+  the desired JSON.
+  Decodin options:
```

Comment:
[nit] `Decodin` -> `Decoding`.

A good idea to split the encoding and decoding options.